app/services/ItemsCachingService.scala

```diff
@@ -0,0 +1,178 @@
+              GoodsMeasure(
+                Some(Measure(value = Some(BigDecimal(form.grossMass)))),
+                Some(Measure(value = Some(BigDecimal(form.netMass)))),
+                Some(Measure(value = form.supplementaryUnits.map((BigDecimal(_)))))
+              )
+            )
+        )
+      )
+
+  def additionalInfo(cachedData: CacheMap) =
+    cachedData
+      .getEntry[AdditionalInformationData](AdditionalInformationData.formId)
+      .map(_.items.map(info => AdditionalInformation(Some(info.code), Some(info.description))))
+
+  def documents(cachedData: CacheMap) =
+    cachedData
+      .getEntry[DocumentsProducedData](DocumentsProducedData.formId)
+      .map(
+        _.documents.map(
+          doc =>
+            GovernmentAgencyGoodsItemAdditionalDocument(
+              doc.documentTypeCode.map(_.substring(0, 1)),
+              typeCode = doc.documentTypeCode.map(_.substring(1)),
+              id = doc.documentIdentifier.map(_ + doc.documentPart.getOrElse("")),
+              lpcoExemptionCode = doc.documentStatus,
+              name = doc.documentStatusReason,
+              writeOff = Some(WriteOff(Some(Measure(value = doc.documentQuantity.map(BigDecimal(_))))))
+          )
+        )
+      )
+
+  def goodsItemFromItemTypes(cachedData: CacheMap) =
+    cachedData
+      .getEntry[ItemType](ItemType.id)
+      .map(
+        item => // get all codes create classification
+          GovernmentAgencyGoodsItem(
+            sequenceNumeric = 1,
+            statisticalValueAmount = Some(Amount(value = Some(BigDecimal(item.statisticalValue)))),
+            commodity = Some(
+              Commodity(
+                description = Some(item.descriptionOfGoods),
+                classifications = Seq(
+                  Classification(
+                    Some(item.combinedNomenclatureCode),
+                    identificationTypeCode = Some(CombinedNomenclatureCode)
+                  ),
+                  Classification(item.cusCode, identificationTypeCode = Some(CUSCode))
+                ) ++
+                  item.nationalAdditionalCodes.map(
+                    code => Classification(Some(code), identificationTypeCode = Some(NationalAdditionalCode))
+                  ) ++ item.taricAdditionalCodes
+                  .map(code => Classification(Some(code), identificationTypeCode = Some(TARICAdditionalCode)))
+              )
+            )
+        )
+      )
+
+  private def createGoodsItem(seq: Int, cachedData: CacheMap) = {
+    val itemTypeData = goodsItemFromItemTypes(cachedData)
+    val commodity = itemTypeData.fold(Commodity())(_.commodity.getOrElse(Commodity()))
+    val updatedCommodity = commodity.copy(goodsMeasure = commodityGoodsMeasure(cachedData).flatMap(_.goodsMeasure))
+
+    GovernmentAgencyGoodsItem(
+      sequenceNumeric = seq,
+      statisticalValueAmount = itemTypeData.flatMap(_.statisticalValueAmount),
+      packagings = generatePackages(cachedData).getOrElse(Seq.empty),
```

Comment:
You can avoid `getOrElse(Seq.empty)` if you solve return types for methods.
It makes unnecessary noise in the code.

Reply:
the getOrElse are there to understand the else scenario to later add validation in all those circumstances where it is None